addons/account_generator/models/generator.py

```diff
@@ -0,0 +1,212 @@
+# -*- coding: utf-8 -*-
+
+from odoo import api, models, fields, tools, _
+from odoo.exceptions import UserError
+
+import random
+import datetime
+import logging
+import datetime
+try:
+    from faker import Faker
+except ImportError:
+    raise UserError('You need to install faker (see https://pypi.org/project/Faker/) in order to install account_generator')
+fake = Faker()
+_logger = logging.getLogger(__name__)
+
+
+class AccountGenerator(models.TransientModel):
+    _name = "account.generator"
+    _description = "Generator"
+
+    generator = fields.Selection([('aml', 'Journal Entry'), ('partner', 'Partner'), ('invoice', 'Invoice'), ('payment', 'Payments'), ('bank_statement', 'Bank Statement')], required=True, default='aml')
+
+    number_to_generate = fields.Integer(default=10)
+    number_of_lines = fields.Integer(default=10)
+    customer = fields.Boolean(default=True)
+    supplier = fields.Boolean(default=True)
+    company_type = fields.Selection([('person', 'Person'), ('company', 'Company')], default='person')
+    post = fields.Boolean()
+    end_date = fields.Date(default=fields.Date.today() + datetime.timedelta(days=30 * 3))
+    start_date = fields.Date(default=fields.Date.today() + datetime.timedelta(days=-365 * 3))
+    empty_field_probability = fields.Float(default=0.7)
+
+    company_id = fields.Many2one('res.company', default=lambda self: self.env.company)
+    partner_ids = fields.Many2many('res.partner', relation="generator_partner_pool")
+    account_ids = fields.Many2many('account.account', relation="generator_account_pool")
+    journal_ids = fields.Many2many('account.journal', relation="generator_journal_pool")
+    partner_excluded_ids = fields.Many2many('res.partner', relation="generator_excluded_partner")
+    account_excluded_ids = fields.Many2many('account.account', relation="generator_excluded_account")
+    journal_excluded_ids = fields.Many2many('account.journal', relation="generator_excluded_journal")
```

Comment:
What's the point of these excluded journals and accounts? I don't see any real added value in using that.